src/EventSubscriber/IndexEventSubscriber.php

```diff
@@ -25,17 +28,21 @@ class IndexEventSubscriber implements EventSubscriberInterface
 {
     private EntityManagerInterface $em;
     private LoggerInterface $logger;
+    private ManagerRegistry $registry;
+    private MetricsService $metricsService;
 
     /**
      * IndexEventSubscriber constructor.
      *
      * @param EntityManagerInterface $entityManager
      * @param LoggerInterface $informationLogger
      */
-    public function __construct(EntityManagerInterface $entityManager, LoggerInterface $informationLogger)
+    public function __construct(EntityManagerInterface $entityManager, LoggerInterface $informationLogger, ManagerRegistry $registry, MetricsService $metricsService)
     {
         $this->em = $entityManager;
         $this->logger = $informationLogger;
+        $this->registry = $registry;
```

Comment:
When injecting the `ManagerRegistry` we shouldn't also inject the `EntityManager`. Instead we should
```php
$this->em = $this->registry->getManager();
```
When we reset on exception, we get a reference to a new instance of the `EntityManager`, not a "reset" of the instance we have, so any reference that is injected by the DI container will then be a obsolete reference. Note the "_You can inject the registry instead to avoid this problem._" part of the doc

```php
    /**
     * Resets a named object manager.
     *
     * This method is useful when an object manager has been closed
     * because of a rollbacked transaction AND when you think that
     * it makes sense to get a new one to replace the closed one.
     *
     * Be warned that you will get a brand new object manager as
     * the existing one is not useable anymore. This means that any
     * other object with a dependency on this object manager will
     * hold an obsolete reference. You can inject the registry instead
     * to avoid this problem.
     *
     * @param string|null $name The object manager name (null for the default one).
     *
     * @return ObjectManager
     */
    public function resetManager($name = null);
```

Reply:
It might be what the docs says, but it's not what works. See the frist comment.

Reply:
The $this->registry->getManager(); returns an object manager object and not an entity manger object.... So that would change data types and autocompleting in IDEs